pkgs/by-name/re/recyclarr/package.nix

```
@@ -0,0 +1,83 @@
+{
+  lib,
+  openssl,
+  writeText,
+  git,
+  buildDotnetModule,
+  dotnetCorePackages,
+  fetchFromGitHub,
+  recyclarr,
+  testers,
+}:
+let
+  nuget-config = writeText "nuget.config" ''
+    <configuration>
+      <packageSources>
+        <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
+      </packageSources>
+    </configuration>
+  '';
+in
+buildDotnetModule rec {
+  pname = "recyclarr";
+  version = "7.2.1";
+
+  src = fetchFromGitHub {
+    owner = "recyclarr";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-KXFGT1fprRKN+V+3k0hpjjaI/xpw6UDAk+jj9zMek7k=";
+  };
+
+  projectFile = "Recyclarr.sln";
+  nugetDeps = ./deps.nix;
+
+  prePatch = ''
+    substituteInPlace src/Recyclarr.Cli/Program.cs \
+      --replace-fail '$"v{GitVersionInformation.SemVer} ({GitVersionInformation.FullBuildMetaData})"' '"${version}-nixpkgs"'
+
+    substituteInPlace src/Recyclarr.Cli/Console/Setup/ProgramInformationDisplayTask.cs \
+      --replace-fail 'GitVersionInformation.InformationalVersion' '"${version}-nixpkgs"'
+  '';
+  patches = [ ./001-Git-Version.patch ];
+
+  dotnetRestoreFlags = [ "--configfile=${nuget-config}" ];
+
+  doCheck = false;
+
+  dotnet-sdk = dotnetCorePackages.sdk_8_0;
+  dotnet-runtime = dotnetCorePackages.runtime_8_0;
+  dotnet-test-sdk = dotnetCorePackages.sdk_8_0;
+
+  executables = [ "recyclarr" ];
+  makeWrapperArgs = [
+    "--prefix PATH : ${
+      lib.makeBinPath [
+        git
+        openssl
+      ]
+    }"
+  ];
+
+  passthru = {
+    updateScript = ./update.sh;
+    tests.version = testers.testVersion { package = recyclarr; };
```

```suggestion
    tests.version = testers.testVersion { package = finalAttrs.finalPackage; };
```

we have finalAttrs support for dotnet since  #331398, please use that here